src/planner/rules/range.rs

```diff
@@ -87,6 +87,10 @@ pub fn analyze_range(egraph: &EGraph, enode: &Expr) -> RangeCondition {
 #[rustfmt::skip]
 pub fn filter_scan_rule() -> Vec<Rewrite> { vec![
     // pushdown range condition to scan
+    rw!("filter-scan-null-eq-true";
+        "(filter ?cond (scan ?table ?columns null))" =>
+        "(filter ?cond (scan ?table ?columns true))"
+    ),
```

Comment:
Should we replace all empty conditions from `null` to `true` again? Because filtering on null is actually equivalent to filtering on false...

Reply:
hmm... really? what's the difference?

I thought null == passing None to the storage iterator filter, and inside the storage implementation null == true.

Reply:
Think about we are going to estimate the number of rows for the scan node:
```
(scan ?table ?columns ?filter)
```

The formula should be:
```
row_count(?table) * selectivity(?filter)
```

Here comes the difference:
```
selectivity(null) == 0
selectivity(true) == 1
```

Reply:
oh so I guess we are messing up with the default condition for storage filter. It should be `true` by default, instead of `null` by default?